Libraries/Components/Button.js

```diff
@@ -132,27 +128,25 @@ class Button extends React.Component<ButtonProps> {
   render(): React.Node {
     const {
       accessibilityLabel,
-      color,
+      color = Platform.OS === 'ios' ? '#007AFF' : '#2196F3',
```

Comment:
### Problem
Specify a default value for props.

### My approach
I might have gone too far but this is a possible solution for this problem. I've added a default value to eligible props at the time of destructuring the props. If user doesn't pass on a value, these defaults would be used according to the platform. It also reduces some hardcoded checks on platform type as in Line 150, 202 and 212.

### Alternatives 
- Can add default value decorator in the comments but it won't be as useful as providing default value right inside the code.

Reply:
Thanks for very thorough work on this! As you've discovered, defaultProps is probably not the way to go, since with functional components they are going to go the way of the dodo.

It's a decision that we have to make, and the tradeoffs as I see are:

- If we annotate this in comments, someone could change the default value in code, and then code and comments are out of sync
- If we annotate this in code, some well-meaning developer might see essentially a "redundant" statement because they know that no default value results to a "falsy" value of `undefined`, and remove the ` = false` default expression, and break the documentation.

There's a general principle of (functional ish) programming that code should be structured in a way that you can look at a function and not need to know anything else beyond the code itself, and its inputs and expected outputs. Everything that's not encoded in that is an implicit dependency, and can become brittle and break when people edit that code without understanding its side effects. 

In this case, removing or adding default props would have a particularly gnarly and unknowable side effect of changing the project's documentation, I'm a bit hesitant to take on this implicit dependency.

So my preference would be to keep just to the `@default` annotation in comments, but I'd also like @motiz88 input on this before making a decision!

Reply:
Yes, Because hardcoding it (as a default) within the props object destructuring is not the standard way, also we are changing the library code which is not necessary here.
We can opt for the annotation so that after applying some scripts we can parse it to generate md doc.

Reply:
@ayushjainrksh I'd like to second @jevakallio's comment, you're being super thorough and it's awesome!

I think for the time being we should limit this project to code changes that don't have any impact on runtime behaviour. It would make reviewing the code much easier and safer when we make PRs to the upstream `react-native` repo.

With that in mind, **I'm fine with a `@default` tag in comments.** Ensuring that this tag is in sync with the actual code would indeed be even better, but let's not solve that now. In the medium term we can reasonably expect people editing a code file to deal with updating both copies of this information.

Note that sometimes a prop already has a value in `defaultProps` or a default initialiser in a functional component - `function MyComponent({someProp = 42}) { /* ... */ }` - and I believe `react-docgen` can already parse, so we might not need to add an explicit `@default` tag. When in doubt, however, let's just add a tag and worry about it later.

(As an aside - deeply analysing statements like `const { color = Platform.OS === 'ios' ? '#007AFF' : '#2196F3' } = this.props` in components can get really complicated really fast, so this is probably not the specific way I'd tackle this problem.)

Reply:
@jevakallio @motiz88 I understand the complications involved, just wanted to run my approach with you to see if that's even possible. Seems like a point of discussion for the future. So I'll just proceed with `@default` tag. Also thanks for clarifying the `defaultProps` being used in some cases. Earlier I was wondering how to tackle those cases with any approach. It seems clear now.